core/matching/matchers/xml_templated_match_test.go

```diff
@@ -0,0 +1,2655 @@
+package matchers
+
+import (
+	"testing"
+
+	. "github.com/beevik/etree"
+	. "github.com/onsi/gomega"
+)
+
+// NOTICE: test_cases advanced feature is comment out temporarily
+// 1. <items x-match="allow-unknown-children"> </items>
+// 2. <item x-match-times="5"></item>
+
+/////////////////////////////////////////
+// isLeaf
+
+func Test_isLeaf_1(t *testing.T) {
+	RegisterTestingT(t)
+	doc := NewDocument()
+	doc.ReadFromString("<xml>aaa</xml>")
+	Expect(isLeaf(doc.Root())).To(BeTrue())
+}
+
+func Test_isLeaf_2(t *testing.T) {
+	RegisterTestingT(t)
+	doc := NewDocument()
+	doc.ReadFromString("<xml></xml>")
+	Expect(isLeaf(doc.Root())).To(BeTrue())
+}
+
+func Test_isLeaf_3(t *testing.T) {
+	RegisterTestingT(t)
+	doc := NewDocument()
+	doc.ReadFromString("<xml />")
+	Expect(isLeaf(doc.Root())).To(BeTrue())
+}
+
+/*
+func Test_isLeaf_4(t *testing.T) {
+	RegisterTestingT(t)
+	doc := NewDocument()
+	doc.ReadFromString("<xml x-match=\"allow-unknown-children\"></xml>")
+	Expect(isLeaf(doc.Root())).To(BeFalse())
+}
+
+func Test_isLeaf_5(t *testing.T) {
+	RegisterTestingT(t)
+	doc := NewDocument()
+	doc.ReadFromString("<xml x-match=\"allow-unknown-children\">aaa</xml>")
+	Expect(isLeaf(doc.Root())).To(BeFalse())
+}
+
+func Test_isLeaf_6(t *testing.T) {
+	RegisterTestingT(t)
+	doc := NewDocument()
+	doc.ReadFromString("<xml x-match-times=\"2\"></xml>")
+	Expect(isLeaf(doc.Root())).To(BeTrue())
+}
+
+func Test_isLeaf_7(t *testing.T) {
+	RegisterTestingT(t)
+	doc := NewDocument()
+	doc.ReadFromString("<xml x-match-times=\"2\">aaa</xml>")
+	Expect(isLeaf(doc.Root())).To(BeTrue())
+}
+*/
+
```

Comment:
why are these ones commented out?

Reply:
Sorry, I forgot to remove them.

It is for partial matching we proposed at first but removed after we noticed some problems in practice as mentioned in my previous comments.

In detail, those test cases are for
- `<list x-match="allow-unknown-children">...</list>`: a list node may contain any nodes we not defined in the matcher explicitly,
- `<item x-match-times="n">...</item>`: an item node template in a list may be repeated by `n` times where `...` can use some regular expression as we discussed above.

We comment out such features instead of removing them directly, because we need keep test case ID (e.g. `7` in `Test_isLeaf_7`) consistent with our Test Case Document . If some test cases are deleted, the IDs become non-continuous and confusing.

Anyway, I will remove those test cases and fix all the test case ID to keep codes clean.

Reply:
ok, lets keep them